src/unwind_dwfl.cc

```diff
@@ -252,35 +253,54 @@ DDRes add_runtime_symbol_frame(UnwindState *us, const Dso &dso, ElfAddress_t pc,
   return add_frame(symbol_idx, map_idx, pc, us);
 }
 
-bool try_attach_module(UnwindState *us, const DsoHdr::DsoMap &map,
-                       bool backpopulate_done) {
+DDRes try_attach_module(UnwindState *us, const DsoHdr::DsoMap &map) {
   bool success = false;
-  for (auto it = map.cbegin(); it != map.cend(); ++it) {
-    const Dso &dso = it->second;
-    if (!dso.is_executable() || dso._type != DsoType::kStandard) {
-      continue;
-    }
-    const FileInfoId_t file_info_id = us->dso_hdr.get_or_insert_file_info(dso);
-    if (file_info_id <= k_file_info_error) {
-      LG_DBG("Unable to find file for DSO %s", dso.to_string().c_str());
-      continue;
-    }
-    const FileInfoValue &file_info_value =
-        us->dso_hdr.get_file_info_value(file_info_id);
-    DDProfMod *ddprof_mod = nullptr;
-    auto res = us->_dwfl_wrapper->register_mod(us->current_ip,
-                                               DsoHdr::get_elf_range(map, it),
-                                               file_info_value, &ddprof_mod);
-    if (IsDDResOK(res)) {
+
+  UniqueFd fd_holder;
+  Elf *elf = nullptr;
+
+  // Find an elf file we can load for this PID
+  for (const auto &[_, dso] : map) {
+    if (dso.is_executable() && dso._type == DsoType::kStandard) {
+      // \fixme{nsavoire}
+      // We don't use dso_hdr.get_or_insert_file_info here
+      // because this would cache the dso file location (/proc/<pid>/root/...)
+      // without registering a dwfl module and later on, if a sample is
+      // processed after the process has exited, the cached location would not
+      // exist anymore and we would fail to register the module (which could
+      // work by looking at / instead of proc/<pid>/root if the location is not
+      // cached).
+      auto file_info = us->dso_hdr.find_file_info(dso);
+      fd_holder = UniqueFd{::open(file_info._path.c_str(), O_RDONLY)};
+      if (!fd_holder) {
+        LG_DBG("Unable to open file for DSO %s", dso.to_string().c_str());
```

Comment:
Maybe something like this ?
```
        LG_DBG("Unable to open file for DSO (PID%d)%s", us->pid,
               dso.to_string().c_str());
...
        LG_WRN("Invalid elf (PID%d)%s", us->pid, file_info._path.c_str());
```
I was getting an invalid elf but I could not find the PID associated.
I think it was because a process was mapping a directory (file was ending with `/`)